plugins/event/sqs/sqs.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package sqs
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/hashicorp/vault/sdk/event"
+	"github.com/hashicorp/vault/version"
+)
+
+var _ event.EventSubscriptionPlugin = (*SQS)(nil)
+
+const pluginType = "sqs"
+
+func New() *SQS {
+	return &SQS{
+		clients:   map[string]*sqs.Client{},
+		queueUrls: map[string]string{},
+	}
+}
+
+type SQS struct {
+	clients    map[string]*sqs.Client
+	queueUrls  map[string]string
+	clientLock sync.RWMutex
+}
+
+func (s *SQS) Initialize(_ context.Context) error {
+	return nil
+}
+
+func (s *SQS) Subscribe(ctx context.Context, request *event.SubscribeRequest) error {
+	var options []func(*config.LoadOptions) error
+	// TODO: support creating the queue
+	accessKey := fmt.Sprintf("%v", request.Config["access_key_id"])
+	secretKey := fmt.Sprintf("%v", request.Config["secret_access_key"])
+	sessionToken := fmt.Sprintf("%v", request.Config["session_token"])
+	region := fmt.Sprintf("%v", request.Config["region"])
+	queueName := fmt.Sprintf("%v", request.Config["queue_name"])
+	if accessKey != "" && secretKey != "" {
+		options = append(options, config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(accessKey, secretKey, sessionToken)))
```

Comment:
I appreciate this is just a draft PR, but just leaving a marker here to remind us to use the AWS auth libraries we already have for a consistent set of supported auth mechanisms.

Reply:
Good call. I'll work on that shortly.